src/main/java/core/basesyntax/service/impl/operation/PurchaseOperation.java

```diff
@@ -0,0 +1,10 @@
+package core.basesyntax.service.impl.operation;
+
+import core.basesyntax.service.OperationHandler;
+
+public class PurchaseOperation implements OperationHandler {
+    @Override
+    public int executeOperation(int initBalance, int quantity) {
+        return initBalance - quantity;
+    }
+}
```

Comment:
The `PurchaseOperation` class does not check if the result balance is positive before returning it. According to the checklist, you should throw a `RuntimeException` if the balance after the purchase operation is negative. This is to prevent the storage from having a negative amount of fruits, which would be an invalid state.

Reply:
++